client/my-sites/plans/components/plans-header/index.tsx

```diff
@@ -64,17 +70,47 @@ const PlansHeader: React.FunctionComponent< {
 		subHeaderText ??
 		translate( 'See and compare the features available on each WordPress.com plan.' );
 
+	const selectedSiteId = useSelector( getSelectedSiteId );
+	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
+	const cart = useShoppingCart( selectedSiteId! );
+
 	if ( domainFromHomeUpsellFlow ) {
 		return <DomainUpsellHeader />;
 	}
 
+	async function handleAddCart() {
+		await cart.addProductsToCart( [
+			{
+				product_slug: PLAN_BUSINESS,
+				meta: 'agency-offer',
+			},
+			{
+				product_slug: PRODUCT_1GB_SPACE,
+				quantity: 50,
+				meta: 'agency-offer',
+			},
+		] );
+
+		window.location.href = `/checkout/${ selectedSiteSlug }`;
+	}
+
 	return (
-		<NavigationHeader
-			className="plans__section-header"
-			navigationItems={ [] }
-			title={ translate( 'Plans' ) }
-			subtitle={ plansDescription }
-		/>
+		<>
+			<NavigationHeader
+				className="plans__section-header"
+				navigationItems={ [] }
+				title={ translate( 'Plans' ) }
+				subtitle={ plansDescription }
+			/>
+			<div style={ { paddingLeft: 32 } }>
+				<Banner
+					title="Bespoke $5/month annual agency Business plan with 100GB storage"
+					primaryButton
+					callToAction="Buy now"
+					onClick={ handleAddCart }
+				/>
+			</div>
```

Comment:
We need to figure out a way of not displaying the offer if...
1. the user is not an agency owner (as defined on the back-end, let's use that as the source of truth)
2. the site already has purchases

Reply:
What I'm thinking is adding a new `GET /sites/%s/special-offers` endpoint to D129558-code. The endpoint can call `is_migration_agency_user()` and if there are special offers it will return the offer text and product slugs. i.e.
```
{
"banner_text": "Bespoke $5/month annual agency Business plan with 100GB storage",
"banner_cta": "Buy now",
"products": [
			{
				product_slug: PLAN_BUSINESS,
				meta: 'agency-offer'
			},
			{
				product_slug: PRODUCT_1GB_SPACE,
				quantity: 50,
				meta: 'agency-offer'
			}
] }
```

That way the backend is deciding on everything to do with the offer.
Unfortunately we'll be adding the network request for every single user even though it'll only be applicable to a few. I think a generic request like `GET all special offers` feels more justifiable than a bespoke request just for our plan.